core/src/main/java/org/lflang/target/property/PlatformProperty.java

```diff
@@ -0,0 +1,262 @@
+import org.lflang.lf.KeyValuePair;
+import org.lflang.lf.KeyValuePairs;
+import org.lflang.lf.LfFactory;
+import org.lflang.lf.LfPackage.Literals;
+import org.lflang.lf.Model;
+import org.lflang.target.TargetProperty;
+import org.lflang.target.property.PlatformProperty.PlatformOptions;
+import org.lflang.target.property.type.DictionaryType;
+import org.lflang.target.property.type.DictionaryType.DictionaryElement;
+import org.lflang.target.property.type.PrimitiveType;
+import org.lflang.target.property.type.TargetPropertyType;
+import org.lflang.target.property.type.UnionType;
+
+public class PlatformProperty extends AbstractTargetProperty<PlatformOptions> {
+
+  public PlatformProperty() {
+    super(UnionType.PLATFORM_STRING_OR_DICTIONARY);
+  }
+
+  @Override
+  public PlatformOptions initialValue() {
+    return new PlatformOptions();
+  }
+
+  @Override
+  public PlatformOptions fromAst(Element node, MessageReporter reporter) {
+    var config = new PlatformOptions();
+    if (node.getLiteral() != null) {
+      config.platform =
+          (Platform) UnionType.PLATFORM_UNION.forName(ASTUtils.elementToSingleString(node));
+      if (config.platform == null) {
+        String s =
+            "Unidentified Platform Type, LF supports the following platform types: "
+                + Arrays.asList(Platform.values());
+        // err.at(value).error(s);
+        throw new AssertionError(s);
+      }
+    } else {
+      for (KeyValuePair entry : node.getKeyvalue().getPairs()) {
+        PlatformOption option =
+            (PlatformOption) DictionaryType.PLATFORM_DICT.forName(entry.getName());
+        switch (option) {
+          case NAME -> {
+            Platform p =
+                (Platform)
+                    UnionType.PLATFORM_UNION.forName(
+                        ASTUtils.elementToSingleString(entry.getValue()));
+            if (p == null) {
+              String s =
+                  "Unidentified Platform Type, LF supports the following platform types: "
+                      + Arrays.asList(Platform.values());
+              reporter.at(entry).error(s);
+              throw new AssertionError(s);
+            }
+            config.platform = p;
+          }
+          case BAUDRATE -> config.baudRate = ASTUtils.toInteger(entry.getValue());
+          case BOARD -> config.board = ASTUtils.elementToSingleString(entry.getValue());
+          case FLASH -> config.flash = ASTUtils.toBoolean(entry.getValue());
+          case PORT -> config.port = ASTUtils.elementToSingleString(entry.getValue());
+          case USER_THREADS -> config.userThreads = ASTUtils.toInteger(entry.getValue());
+          default -> {}
+        }
+      }
+    }
+
+    return config;
+  }
+
+  @Override
+  protected PlatformOptions fromString(String string, MessageReporter reporter) {
+    throw new UnsupportedOperationException("Not supported yet.");
+  }
+
+  @Override
+  public List<Target> supportedTargets() {
+    return Target.ALL;
```

Comment:
afaik, this is only supported by the C target.

Reply:
5d499c8bf